PSL cover - failure

**Description**
When a PSL cover point is not covered during simulation, at the end, error (failure) is reported.

**Expected behaviour**
Uncovered PSL cover points shall not report error (failure), this is also how commercial simulators behave.
The rationale behind is:
 - One test-bench with many cover points scattered through TB and RTL.
 - Many tests. Each test is separate run of simulator (different procedure), but the same test-bench. E.g. as in Vunit.
 - Each test indends to verify (and cover) different scenario, situation.
 - After whole regression, functional coverage from all tests is merged together to find out if there were some cover points
   not hit by any test (e.g. in some HTML report)..

GHDL behavior in this situation makes all test fail (unless there is a test which covers all cover points, which is unrealistic),
making the PSL implementation completely unusable for any decently complex testbench.

**How to reproduce?**


```vhd :file: top.vhd
Library ieee;
use ieee.std_logic_1164.all;

entity tb_top is
end entity;


architecture tb of tb_top is

    signal a,b,c : std_logic;
    signal clk_sys : std_logic;

begin
	
    gen_clock_proc : process
    begin
        clk_sys <= '1';
        wait for 5 ns;
        clk_sys <= '0';
        wait for 5 ns;
    end process;
	
	test_proc : process
	begin
	    a <= '0';
	    b <= '0';
	    c <= '0';
	    wait until rising_edge(clk_sys);
	    a <= '1';
	    wait until rising_edge(clk_sys);
	    a <= '0';
	    b <= '1';
	    wait until rising_edge(clk_sys);
	    b <= '0';
	    c <= '1';
	    wait until rising_edge(clk_sys);
	    
        std.env.finish;
	end process;


    --  psl default clock is rising_edge(clk_sys);

    -- psl cov_simult_a_b_c : cover {a = '1' and b = '1' and c = '1'};

end architecture tb;
```

```sh :image: ghdl/ghdl:buster-gcc-8.3.0
ghdl -a --workdir=build --std=08 -fpsl top.vhd
ghdl -e --workdir=build --std=08 tb_top
ghdl -r tb_top
```

> NOTE: `:file:` and `:image:` identifiers are specific to [issue-runner](https://github.com/1138-4EB/issue-runner). We suggest to use these, since it allows continuous integration workflows to automatically test the MWE. Using `ghdl/ghdl:*` docker images to run the MWEs ensures that the latest available GHDL is used.

> NOTE: Large files can be uploaded one-by-one or in a tarball/zipfile.

**Context**
Please, provide the following information:

- OS: Ubuntu
- Origin:
  - [ ] Package manager: `version`
  - [ ] Released binaries: `tarball_url`
  - [x] Built from sources: 8eb6eb35ae475be271cef614af0256282286606b

**Additional context**

Simulation log:
simulation finished @40ns
top.vhd:45:12:@40ns:(psl cover failure error): sequence not covered

I didn't know that PSL cover directives are also taken in account when using the `--assert-level=` option. I could reproduced your problem:

```shell
cd work/psl_cover; ghdl -r --std=08 -fpsl tb_psl_cover --assert-level=warning --wave=psl_cover.ghw --psl-report=psl_cover_psl_coverage.json
../src/psl_cover.vhd:32:3:@2ns:(psl cover note): Transfer requested
../src/psl_cover.vhd:37:3:@3ns:(psl cover note): Transfer in progress
../src/psl_cover.vhd:37:3:@4ns:(psl cover note): Transfer in progress
../src/psl_cover.vhd:76:3:@8ns:(psl cover note): Transfer of length 3
../src/psl_cover.vhd:42:3:@9ns:(psl cover note): Transfer done
simulation finished @12ns with status 0
../src/psl_cover.vhd:56:3:@12ns:(psl cover failure error): sequence not covered
./tb_psl_cover:error: psl cover failure failed
in process .tb_psl_cover(sim).dut@psl_cover(psl).P1
make: *** [Makefile:17: psl_cover] Error 1
```

I agree that this isn't that useful. Initially I thought you only had the warnings in the report, but with that option teh simulation also fails when cover points weren't hit during simulation. So I also vote for a runtime option which explicitly enables the severity settings for PSL cover directives.

--- a/doc/using/Simulation.rst
+++ b/doc/using/Simulation.rst
@@ -338,3 +338,7 @@ Export hierarchy and references
 .. option:: --psl-report=<FILENAME>
 
   Write a report for PSL at the end of simulation. For each PSL cover and assert statements, the name, source location and whether it passed or failed is reported. The file is written using the JSON format, but is still human readable.
+
+.. option:: --psl-report-uncovered
+
+  Reports warning for each uncovered PSL cover point when simulation ends.

